Formula/v8.rb

```diff
@@ -3,110 +3,90 @@
-    (buildpath/"third_party/icu").install resource("icu")
-    (buildpath/"buildtools").install resource("buildtools")
-    (buildpath/"base/trace_event/common").install resource("common")
-    (buildpath/"tools/swarming_client").install resource("swarming_client")
-    (buildpath/"testing/gtest").install resource("gtest")
-    (buildpath/"testing/gmock").install resource("gmock")
-    (buildpath/"tools/clang").install resource("clang")
-
-    system "make", "native", "library=shared", "snapshot=on",
-                   "console=readline", "i18nsupport=off",
-                   "strictaliasing=off"
-
-    include.install Dir["include/*"]
-
-    cd "out/native" do
-      rm ["libgmock.a", "libgtest.a"]
-      lib.install Dir["lib*"]
-      bin.install "d8", "mksnapshot", "process", "shell" => "v8"
+    (buildpath/"depot_tools").install resource("depot_tools")
+    ENV.prepend_path "PATH", buildpath/"depot_tools"
+
+    # This env variable used by gclient to prevent depot_tools to update depot_tools on every call
+    # see https://www.chromium.org/developers/how-tos/depottools#TOC-Disabling-auto-update
+    ENV["DEPOT_TOOLS_UPDATE"] = "0"
+
+    repo_cache = HOMEBREW_CACHE/"#{name}--v8--gclient/"
+    repo_cache.mkpath
+
+    # Configure build
+    gn_args = {
+      :is_debug => false,
+      :is_component_build => true,
+      :v8_use_external_startup_data => false,
+    }
+
+    v8_version = version
+    arch = MacOS.prefer_64_bit? ? "x64" : "x86"
+    output_name = "#{arch}.release"
+    output_path = "out.gn/#{output_name}"
+
+    gn_command = "gn gen #{output_path} --args=\"#{gn_args.map { |k, v| "#{k}=#{v}" }.join(" ")}\""
+
+    cd repo_cache do
+      system "gclient", "root"
+      system "gclient", "config", "--spec", <<~EOS
+        solutions = [
+          {
+            "url": "https://chromium.googlesource.com/v8/v8.git",
+            "managed": False,
+            "name": "v8",
+            "deps_file": "DEPS",
+            "custom_deps": {},
+          },
+        ]
+        target_os = [ "mac" ]
+        target_os_only = True
+      EOS
+
+      system "gclient", "sync", "-vvv", "-j #{Hardware::CPU.cores}", "-r", v8_version
+
+      cd "v8" do
+        system gn_command
+        system "ninja", "-j #{Hardware::CPU.cores}", "-v", "-C", output_path, "d8"
+
+        include.install Dir["include/*"]
+
+        cd output_path do
+          lib.install Dir["lib*.dylib", "icudtl.dat", "d8"]
+        end
+      end
     end
   end
 
   test do
-    assert_equal "Hello World!", pipe_output("#{bin}/v8 -e 'print(\"Hello World!\")'").chomp
+    test_basic_script = <<-EOS.strip
+      print("Hello World!");
+    EOS
+
+    test_icu_script = <<-EOS.undent.tr("\n", " ").strip
+      var date = new Date(Date.UTC(2012, 11, 20, 3, 0, 0));
+      print(new Intl.DateTimeFormat("en-US").format(date));
+    EOS
+
+    assert_equal "Hello World!", pipe_output("#{lib}/d8 -e '#{test_basic_script}'").chomp
```

Comment:
`pipe_output` doesn't bother checking the return code by default, so we should handle that here to assure we're getting back `0`.

Reply:
I switched to `shell_output`.